plugins/bazaar/src/components/EntityBazaarInfoCard/EntityBazaarInfoCard.tsx

```diff
@@ -0,0 +1,345 @@
+
+  const getInitMemberStatus = async () => {
+    const response = await bazaarApi.getMembers(entity);
+    const dbMembers = response.data.map((obj: any) => {
+      const member: Member = {
+        userId: obj.user_id,
+        entityRef: obj.entity_ref,
+        joinDate: obj.join_date,
+      };
+
+      return member;
+    });
+
+    dbMembers.sort(sortMembers);
+
+    setMembers(dbMembers);
+    setIsMember(
+      dbMembers
+        .map((member: Member) => member.userId)
+        .indexOf(identity.getUserId()) >= 0,
+    );
+  };
+
+  const getMetadata = async () => {
+    const response = await bazaarApi.getMetadata(entity);
+
+    if (response.status !== 404) {
+      setIsBazaar(true);
+      const data = await response.json().then((resp: any) => resp.data);
+
+      setBazaarProject({
+        entityRef: data[0].entityRef,
+        name: data[0].name,
+        community: data[0].community,
+        announcement: data[0].announcement,
+        status: data[0].status,
+        updatedAt: data[0].updatedAt,
+      });
+    }
+  };
+
+  const { loading } = useAsync(async () => {
+    await getInitMemberStatus();
+    await getMetadata();
+  });
+
+  const onOpen = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+    setPopoverOpen(true);
+  };
+
+  const closeEdit = () => {
+    setOpen(false);
+  };
+
+  const closeDelete = () => {
+    setOpenDelete(false);
+  };
+
+  const popoverCloseHandler = () => {
+    setPopoverOpen(false);
+  };
+
+  const handleMembersClick = async () => {
+    setIsMember(!isMember);
+
+    const newMember: Member = {
+      userId: identity.getUserId(),
+      entityRef: stringifyEntityRef(entity),
+      joinDate: new Date().toISOString(),
+    };
+
+    if (!isMember) {
+      setMembers((prevMembers: Member[]) => {
+        const newMembers: Member[] = [...prevMembers, newMember];
+        newMembers.sort(sortMembers);
+        return newMembers;
+      });
+      await bazaarApi.addMember(entity);
+    } else {
+      setMembers(
+        members.filter(
+          (member: Member) => member.userId !== identity.getUserId(),
+        ),
+      );
+      await bazaarApi.deleteMember(entity);
+    }
+  };
+
+  const links: IconLinkVerticalProps[] = [
+    {
+      label: 'Community',
+      icon: <ChatIcon />,
+      href: bazaarProject.community,
+      disabled: bazaarProject.community === '',
+    },
+    {
+      label: isMember ? 'Leave' : 'Join',
+      icon: isMember ? <ExitToAppIcon /> : <PersonAddIcon />,
+      href: '',
+      onClick: async () => {
+        handleMembersClick();
+      },
+    },
+  ];
+
+  if (loading) {
+    return <Progress />;
+  } else if (!isBazaar) {
+    return (
+      <Card>
+        <CardHeader title="Bazaar" style={{ paddingBottom: '1rem' }} />
+        <Divider />
+        <CardContent>
+          <Typography variant="body1">
+            This project is not in the Bazaar. Go to the{' '}
```

Comment:
So, I'm wondering, is it going to be the case that the vast majority of entities would have a bazaar entry? I'm just asking because otherwise, if one puts this card on the overview page, it'll load and show a progress bar and kinda show this announcement almost every single time that you go anywhere. Just a bit unfortunate if "the normal state" of things would be to show that there's nothing to show. Not sure how to best address that though, I understand that one wants some kind of a "hook" into the bazaar system too; the alternative is to not show the card at all when there's no bazaar info related to the entity.

Reply:
We tried to not show the Bazaar card on the entity page if the project is not in the bazaar without any success. The problem was that we need to fetch and use an async function to determine whether or not the card should be shown, but we can't pass an async function to the EntitySwitch.Case. We asked about this in Discord and got the suggestion that we always could show the card. Do you have any suggestions to how we could fix this, because as you mention, most of the projects will not be in the bazaar and that is the solution we were aiming for.